app/controllers/api/api_controller.rb

```diff
@@ -445,8 +445,11 @@ def pept2interpro_helper
 
           v[:ipr].each do |value|
             ipr_entry = ipr_mapping[value[:code]]
-            value[:name] = ipr_entry.name if @extra_info
-            splitted[ipr_entry.category] << value
+
+            if !ipr_entry.nil?
+              value[:name] = ipr_entry.name if @extra_info
```

Comment:
shouldn't this also be an empty string if the entry is nil?

Reply:
If the entry is nil, the value is not added to the resulting output (splitted). We need to know what the category of the value is, in order to correctly add it to the domain to which it belongs. If we decide to include it in the output here, we'll have an empty string as the key of some JSON-object in the output, which leads to problems:

e.g.:

```json
{
"ipr": 
    [
            {
                "": [],
                "Domain": []
            }
    ]
}
```